src/Orleans.BroadcastChannel/ChannelId.cs

```diff
@@ -164,35 +164,33 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
         }
 
         /// <inheritdoc/>
-        public override string ToString()
-        {
-            var key = this.GetKeyAsString();
-            return keyIndex == 0 ? "null/" + key : this.GetNamespace() + "/" + key;
-        }
+        public override string ToString() => $"{this}";
+        string IFormattable.ToString(string? format, IFormatProvider? formatProvider) => ToString();
 
-        /// <summary>
-        /// Parses a <see cref="ChannelId"/> instance from a <see cref="string"/>.
-        /// </summary>
-        /// <param name="value">The value.</param>
-        /// <returns>The parsed stream identity.</returns>
-        public static ChannelId Parse(string value)
+        bool ISpanFormattable.TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan<char> format, IFormatProvider? provider)
         {
-            if (string.IsNullOrWhiteSpace(value))
+            var len = Encoding.UTF8.GetCharCount(fullKey);
+            if (keyIndex == 0)
             {
-                ThrowInvalidInternalStreamId(value);
+                if (destination.Length > len + 5)
```

Comment:
Should this be `>=`? Eg, imagine `fullKey` is a single byte, `'x'`, then `"null/x".Length == 6`, which will fit in `destination` as long as its length is greater or equal to 6